src/utils/simulation_results.jl

```diff
@@ -164,16 +164,25 @@ function check_file_integrity(path::String)
         expected_hash, file_name = split(line)
         actual_hash = compute_sha256(file_name)
         if expected_hash != actual_hash
-            @error "hash mismatch for file" file_name expected_hash actual_hash
+            @debug "hash mismatch for file" file_name expected_hash actual_hash
             matched = false
-        else
-            @info "File hash values matched." file_name
         end
     end
 
     if !matched
-        throw(IS.DataFormatError(
+        throw(IS.HashMismatchError(
             "The hash value in the written files does not match the read files, results may have been tampered."
         ))
     end
 end
+
+function get_variable_names(sim::Simulation, stage::Any)
+     return collect(keys(sim.stages[stage].internal.psi_container.variables))
+ end
+ function get_reference(sim_results::SimulationResultsReference, stage::String, step::Int, variable::Symbol) 
+     file_paths = sim_results.ref["stage-$stage"][variable]
+     file_paths = filter(file_paths -> file_paths.Step == "step-$step", file_paths)[:, 3]
+     return file_paths
+ end
+
+  get_psi_container(sim::Simulation, stage::Any) = sim.stages[stage].internal.psi_container
```

Comment:
Would it be better to check if `stage` is present before trying to dereference it?

Reply:
if stage isn't present a KeyError already gets thrown

Reply:
This was defined as an exported function.  Getting a KeyError on an exported function exposes underlying implementation details to the user, which you wouldn't want to do.  If the function is internal only then it doesn't really matter.

Reply:
This shouldn't be an exported function.